app/javascript/selectors/detailSelector.test.js

```diff
@@ -369,6 +375,40 @@ describe('selectors', () => {
     })
   })
 
+  describe('#isEmailValid', () => {
+    it('return true when email is valid', () => {
+      const state = getState({ email: 'Hello@gmail.com' })
+      expect(isEmailValid(state)).toEqual(true)
+    })
+
+    it('return false if email is not valid', () => {
+      const state = getState({ email: 'hello@' })
+      expect(isEmailValid(state)).toEqual(false)
+    })
+  })
+
+  describe('#disableActionButton', () => {
+    it('return true when email is valid and disableActionBtn is true', () => {
+      const state = getState({ email: 'Hello@gmail.com', disableActionBtn: true })
+      expect(disableActionButton(state)).toEqual(true)
+    })
+
+    it('return true when email is not valid and disableActionBtn is false', () => {
+      const state = getState({ email: 'hello@', disableActionBtn: false })
+      expect(disableActionButton(state)).toEqual(true)
+    })
+
+    it('return false when email is valid and disableActionBtn is false', () => {
+      const state = getState({ email: 'hello@gmail.com', disableActionBtn: false })
+      expect(disableActionButton(state)).toEqual(false)
+    })
+
+    it('return true when email is not valid and disableActionBtn is true', () => {
```

Comment:
You could remove his and simplify the it block on line 396 to just :
```suggestion
  it('return true when email is not valid', () => {
       const state = getState({ email: 'hello@', disableActionBtn: false })
       expect(disableActionButton(state)).toEqual(true)
     })
```

Reply:
because overall any invalid email must cause us to return true